clients/src/main/java/org/apache/kafka/common/requests/DeleteGroupsResponse.java

```diff
@@ -16,121 +16,72 @@
  */
 package org.apache.kafka.common.requests;
 
+import org.apache.kafka.common.message.DeleteGroupsResponseData;
+import org.apache.kafka.common.message.DeleteGroupsResponseData.DeletableGroupResult;
 import org.apache.kafka.common.protocol.ApiKeys;
 import org.apache.kafka.common.protocol.Errors;
-import org.apache.kafka.common.protocol.types.ArrayOf;
-import org.apache.kafka.common.protocol.types.Field;
-import org.apache.kafka.common.protocol.types.Schema;
 import org.apache.kafka.common.protocol.types.Struct;
 
 import java.nio.ByteBuffer;
-import java.util.ArrayList;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
-import static org.apache.kafka.common.protocol.CommonFields.ERROR_CODE;
-import static org.apache.kafka.common.protocol.CommonFields.GROUP_ID;
-import static org.apache.kafka.common.protocol.CommonFields.THROTTLE_TIME_MS;
-
 public class DeleteGroupsResponse extends AbstractResponse {
-    private static final String GROUP_ERROR_CODES_KEY_NAME = "group_error_codes";
-
-    private static final Schema GROUP_ERROR_CODE = new Schema(
-            GROUP_ID,
-            ERROR_CODE);
-
-    private static final Schema DELETE_GROUPS_RESPONSE_V0 = new Schema(
-            THROTTLE_TIME_MS,
-            new Field(GROUP_ERROR_CODES_KEY_NAME, new ArrayOf(GROUP_ERROR_CODE), "An array of per group error codes."));
-
-    /**
-     * The version number is bumped to indicate that on quota violation brokers send out responses before throttling.
-     */
-    private static final Schema DELETE_GROUPS_RESPONSE_V1 = DELETE_GROUPS_RESPONSE_V0;
-
-    public static Schema[] schemaVersions() {
-        return new Schema[]{DELETE_GROUPS_RESPONSE_V0, DELETE_GROUPS_RESPONSE_V1};
-    }
 
+    public final DeleteGroupsResponseData data;
 
-    /**
-     * Possible error codes:
-     *
-     * COORDINATOR_LOAD_IN_PROGRESS (14)
-     * COORDINATOR_NOT_AVAILABLE(15)
-     * NOT_COORDINATOR (16)
-     * INVALID_GROUP_ID(24)
-     * GROUP_AUTHORIZATION_FAILED(30)
-     * NON_EMPTY_GROUP(68)
-     * GROUP_ID_NOT_FOUND(69)
-     */
-
-    private final Map<String, Errors> errors;
-    private final int throttleTimeMs;
-
-    public DeleteGroupsResponse(Map<String, Errors> errors) {
-        this(DEFAULT_THROTTLE_TIME, errors);
-    }
-
-    public DeleteGroupsResponse(int throttleTimeMs, Map<String, Errors> errors) {
-        this.throttleTimeMs = throttleTimeMs;
-        this.errors = errors;
+    public DeleteGroupsResponse(DeleteGroupsResponseData data) {
+        this.data = data;
     }
 
     public DeleteGroupsResponse(Struct struct) {
```

Comment:
Do we need this Constructor? It looks like it's only called from `parse()` which has the version so it could call the other constructor

Reply:
Yea, I think it's ok, some other test like `RequestQuotaTest` also use it.